### ISSUE TITLE
Rely: Float Matcher

### ISSUE THREAD

**Issue by bryphe** [facebookexperimental/reason-native#39]
I'm working on switching to `rely` for my `revery` project: https://github.com/bryphe/revery/pull/100

One blocker I'm hitting now is that there is no `float` matcher. The tricky thing about floats is that often you don't want `==` semantics - you want to verify that the value is within a delta.

I'd suggest a similiar matcher to `expect.int`, and have `toBe` implement `==` semantics, and `toBeCloseTo(~delta:float?, v)`, where `toBeCloseTo` checks if the floats are within a delta (optionally specified by the `~delta` parameter), ie:

```
expect.float(0.001).toBeCloseTo(0.00100000000173) // TRUE
```

The default delta could be something like `0.00001`.

Jest uses an alternative to `delta` - they use `numDigits` (with a default of 2), which would work fine too.

**Comment by bryphe** [comment]
BTW, I'm happy to help out with this, just don't want to overlap work if you're already thinking / working on this

**Comment by kyldvs** [comment]
Yeah we have something really similar to Jest with a default of 2. We actually already built this in the js-compiled version of Rely just need to add it here:

```
test("float utils", ({expect}) => {
  expect.float(0.1 +. 0.2).toBeCloseTo(0.3);
  expect.float(0.1 +. 0.2).toBeCloseTo(~digits=15, 0.3);
  expect.float(0.1 +. 0.2).not.toBeCloseTo(0.4);
  expect.float(0.1 +. 0.2).not.toBeCloseTo(~digits=16, 0.3);
});
```

**Comment by kyldvs** [comment]
And if you want to help out and add this that would be great! I don't believe anyone is actively adding this yet. We will soon though if needed

**Comment by bandersongit** [comment]
PR is out https://github.com/facebookexperimental/reason-native/pull/67. Went with the jest approach/what Kyle alluded to us having internally

**Comment by bandersongit** [comment]
The PR was merged yesterday and float matchers are included in version 1.0 of Rely which I just published to npm.
